core/app/models/spree/shipping_method.rb

```diff
@@ -41,7 +41,7 @@ class ShippingMethod < Spree::Base
 
     scope :available_to_store, ->(store) do
       raise ArgumentError, "You must provide a store" if store.nil?
-      store.shipping_methods.empty? ? all : where(id: store.shipping_method_ids)
+      store.shipping_methods.empty? ? all : where(id: store.shipping_methods.ids)
```

Comment:
What's the difference here? Shouldn't they return the same thing?

Reply:
They should; I checked the generated SQL on the sandbox and the output was the same for both calls, but somehow, the result is not the same. (at least, not in this scenario.)

I will investigate further to understand what's going on here.

Reply:
@kennyadsl I added this:

```ruby
puts store.shipping_method_ids.inspect
```

to the scope and found that `store.shipping_method_ids` is returning **just one** shipping method ID, whereas `store.shipping_methods.ids` is returning **all** the IDs for the shipping methods associated to the store.

Can't understand why this is happening, as the generated SQL is exactly the same for both, and some unit tests I created for this work as expected with either. Maybe there's a bug present in Rails (as mentioned first by @afdev82 in [here](https://github.com/solidusio/solidus/issues/2998#issuecomment-446933585)) or something else we're missing out.

Reply:
Trying this out and it returns **just one** shipping method ID also for:

```ruby
ActiveRecord::Base.uncached { store.shipping_method_ids } # => [3]
```

in the test you wrote.

Also, the query seems a bit too much for what it should do:

```sql
(0.6ms)
SELECT DISTINCT "spree_shipping_methods"."id"
FROM "spree_shipping_methods"
LEFT OUTER JOIN "spree_shipping_method_stock_locations" ON "spree_shipping_methods"."id" = "spree_shipping_method_stock_locations"."shipping_method_id"
INNER JOIN "spree_store_shipping_methods" ON "spree_shipping_methods"."id" = "spree_store_shipping_methods"."shipping_method_id"
WHERE "spree_shipping_methods"."deleted_at" IS NULL
  AND "spree_shipping_methods"."id" IN
    (SELECT "spree_shipping_methods"."id"
     FROM "spree_shipping_methods"
     INNER JOIN "spree_shipping_method_categories" ON "spree_shipping_method_categories"."shipping_method_id" = "spree_shipping_methods"."id"
     WHERE "spree_shipping_methods"."deleted_at" IS NULL
       AND "spree_shipping_method_categories"."shipping_category_id" = ?
     GROUP BY spree_shipping_methods.id
     HAVING COUNT(DISTINCT "spree_shipping_method_categories"."id") = 1)
  AND ("spree_shipping_methods"."available_to_all" = 1
       OR "spree_shipping_method_stock_locations"."stock_location_id" = 1)
  AND "spree_store_shipping_methods"."store_id" = ? [["shipping_category_id", 2],
      ["store_id", 1]]
```

There are some suspect join with categories, which I'm not sure why are there and if they are needed here. Can you please investigate a little bit more?

Reply:
Yeah sure, let me have a further look at it